### PR TITLE
Devtools styles & styled components fix

### PR DESCRIPTION
**Pull request by andrejak**
Fixes a couple issues with devtools and an app using styled components.
- make todo-app use styled-components for easier testing
- ensure versions are consistent and types are set to 4.1.8 everywhere to prevent these warnings:
Screenshot 2019-09-26 at 12 47 59
- move style copying code because it wasn't copying the app's styled components
_(there are probably some other issues with the copying that I'll raise and handle separately - eg we might accidentally apply some of users' styles to devtools...)_

End result seems to be working:
Screenshot 2019-09-26 at 16 15 56

### CODE REVIEW

File: examples/todo-app/package.json [hunk 1]
```diff
@@ -16,7 +16,8 @@
     "@prodo/logger": "^0.0.9",
     "core-js": "2",
     "react": "^16.8.6",
-    "react-dom": "^16.9.0"
+    "react-dom": "^16.9.0",
+    "styled-components": "^4.4.0"
```
Comment (coffee-cup) [inline examples/todo-app/package.json]: If the versions are different does it break?

Comment (andrejak) [reply examples/todo-app/package.json]: Actually, it seems to be fine now. I thought that was causing issues initially but I think it was mostly the types + that last commit. (If you want me to change the docs or todo-app back to 4.3.2 or something, lmk)

Comment (coffee-cup) [reply examples/todo-app/package.json]: This version is okay. I was just didn't think we should be forcing users to use a specific version of styled-components. If multiple versions are allowed then that is good 👍

Comment (coffee-cup) [reply examples/todo-app/package.json]: I do remember having a bunch of issues with `@types/styled-components`

File: packages/devtools-core/src/App/UserAppContainer/index.tsx [hunk 2]
```diff
@@ -49,20 +57,10 @@ const UserAppContainer = (props: Props) => {
         }
       };
     }
-  }, []);
+  });
```
Comment (coffee-cup) [inline packages/devtools-core/src/App/UserAppContainer/index.tsx]: Why remove the empty array? Is this effect supposed to be run whenever any prop changes?

Comment (andrejak) [reply packages/devtools-core/src/App/UserAppContainer/index.tsx]: When it runs for the first time, styled components aren't ready yet, so they don't get copied over. But you're right, the listeners should only run once - will separate the effects.

**Comment by andrejak** [comment]
Realized I had an idea about how to fix the copying issue I mentioned above, so I added this to this PR - hope that's ok.

**Comment by andrejak** [comment]
With the above we might need to add the css mocking config to example templates... Lmk if this is not a good idea.

**Comment by coffee-cup** [comment]
@andrejak definitely a good idea to add mocking in templates. I can add that in webpack update.

File: packages/devtools-core/src/App/UserAppContainer/index.tsx (hunk 2 above)
Comment (coffee-cup) [reply packages/devtools-core/src/App/UserAppContainer/index.tsx]: Are the style nodes going to continuously be added every time this component rerenders? Maybe they should be removed in the return callback of use effect?

Comment (andrejak) [reply packages/devtools-core/src/App/UserAppContainer/index.tsx]: Just noting here for posterity - the styled components style nodes do keep being added. For now this seems to work fine when I edit stylesheets or styled components. If we run into issues with the future, we should revisit.

**Pull request merged**